circuit_knitting/cutting/cutting_experiments.py

```diff
@@ -273,19 +305,29 @@ def _append_measurement_circuit(
                 f"qubit_locations has {len(qubit_locations)} element(s) but the "
                 f"observable(s) have {cog.general_observable.num_qubits} qubit(s)."
             )
+
+    # Find observable_measurements register
+    for reg in qc.cregs:
+        if reg.name == "observable_measurements":
+            obs_creg = reg
+            break
+    else:
+        raise ValueError('Cannot locate "observable_measurements" register')
+
+    pauli_indices = _get_pauli_indices(cog)
+
+    if obs_creg.size != len(pauli_indices):
+        raise ValueError(
+            '"observable_measurements" register is the wrong size '
+            "for the given commuting observable group "
+            f"({obs_creg.size} != {len(pauli_indices)})"
+        )
+
     if not inplace:
         qc = qc.copy()
 
-    # If the circuit has no measurements, the Sampler will fail.  So, we
```

Comment:
Could you quickly explain why we can remove this? I may figure it out as I continue the review, but just in case ;)

Reply:
Got moved to a new function `_get_pauli_indices`, because it needs checked now in two places. :)